src/fallback.ts

```diff
@@ -313,30 +325,90 @@ export class GrpcClient {
         const protoServiceName = protoNamespaces.join('.');
         const rpcName = method.name;
 
-        const url = `${grpcFallbackProtocol}://${servicePath}:${servicePort}/$rpc/${protoServiceName}/${rpcName}`;
+        let url: string;
+        let data: string;
+        let httpMethod: string;
+
+        if (this.fallback === 'json') {
+          // REGAPIC: JSON over HTTP/1 with gRPC trancoding
+          headers['Content-Type'] = 'application/json';
+          const decodedRequest = method.resolvedRequestType.decode(requestData);
+          const requestJSON = method.resolvedRequestType.toObject(
+            // TODO: use toJSON instead of toObject
+            decodedRequest
+          );
+          const transcoded = transcode(requestJSON, method.parsedOptions);
+          if (!transcoded) {
+            throw new Error(
+              `Cannot build HTTP request for ${JSON.stringify(
+                requestJSON
+              )}, method: ${method.name}`
+            );
+          }
+          httpMethod = transcoded.httpMethod;
+          data = JSON.stringify(transcoded.data);
+          url = `${grpcFallbackProtocol}://${servicePath}:${servicePort}/${transcoded.url.replace(
+            /^\//,
+            ''
+          )}?${transcoded.queryString}`;
+        } else {
+          // gRPC-fallback: proto over HTTP/1
+          headers['Content-Type'] = 'application/x-protobuf';
+          httpMethod = 'post';
+          data = requestData;
+          url = `${grpcFallbackProtocol}://${servicePath}:${servicePort}/$rpc/${protoServiceName}/${rpcName}`;
+        }
 
         const fetch = isBrowser()
           ? // eslint-disable-next-line no-undef
             window.fetch
           : ((nodeFetch as unknown) as NodeFetchType);
-        fetch(url, {
+        const fetchRequest = {
           headers,
-          method: 'post',
-          body: requestData,
+          body: data,
+          method: httpMethod,
           signal: cancelSignal,
-        })
+        };
+        if (
+          httpMethod === 'get' ||
+          httpMethod === 'delete' ||
```

Comment:
any concern `httpMethod` could be `GET` or `POST`, is it worth lowercasing, or do we always control the value?

Reply:
We control it here (they come from `transcode()` in `transcoding.ts`) so we know for sure they are lower cased. They actually come from `http.proto` where they are lower cased too.